### PR TITLE
Refactor fault exception.

### PR DESCRIPTION
**Pull request by piotrooo**
To issue #53

### CODE REVIEW

File: src/fXmlRpc/Parser/NativeParser.php [hunk 1]
```diff
@@ -26,13 +26,12 @@
 use DateTime;
 use DateTimeZone;
 use fXmlRpc\Exception\MissingExtensionException;
+use fXmlRpc\Exception\ResponseException;
```
Comment (lstrojny) [inline src/fXmlRpc/Parser/NativeParser.php]: I wonder if this should now be a ParseException, as the parser shouldn't need to know anything about the response

**Comment by lstrojny** [comment]
Looks good, just one question regarding `ResponseException`. Could you update the README to cover the change in the 0.20 (as the interface changes this will be the next version) and include it in https://github.com/lstrojny/fxmlrpc#latest-improvements.

File: src/fXmlRpc/Parser/NativeParser.php (hunk 1 above)
Comment (piotrooo) [inline src/fXmlRpc/Parser/NativeParser.php]: `ParseException` suggest that is some kind of error while parsing, but isn't, this is normal behavior of server to sending fault, maybe some kind of `FaultWrapperException`.

Comment (lstrojny) [inline src/fXmlRpc/Parser/NativeParser.php]: Just FaultException?

Comment (piotrooo) [inline src/fXmlRpc/Parser/NativeParser.php]: Good enough for me.

Comment (lstrojny) [inline src/fXmlRpc/Parser/NativeParser.php]: Alright, let’s go with it

**Comment by piotrooo** [comment]
Done.

**Comment by lstrojny** [comment]
Could you add the migration path in the README for 0.20.0 so that a user who implemented his owner parse implementation is not confused? If you squash afterwards we are good to go

**Comment by lstrojny** [comment]
Sorry this took so incedibly long. Could you rebase and squash so we can merge?

**Comment by lstrojny** [comment]
Something doesn’t look good here. Can you open a new PR or whatever is the easiest or you.

**Pull request closed without merging**